src/api/init.ts

```diff
@@ -224,6 +224,9 @@ export async function startApiServer(opts: {
 
   // Setup error handler (must be added at the end of the middleware stack)
   app.use(((error, req, res, next) => {
+    if (error instanceof RespError) {
```

Comment:
This should be moved further down, next to the existing `if (error && !res.headersSent)` condition.

If headers have already been sent, then this will fail.